packages/sdk/src/makeStreamRpcClient.ts

```diff
@@ -70,3 +73,68 @@ export function getMaxTimeoutMs(opts: StreamRpcClientOptions): number {
     }
     return maxTimeoutMs
 }
+
+export async function getMiniblocks(
+    client: StreamRpcClient,
+    streamId: string | Uint8Array,
+    fromInclusive: bigint,
+    toExclusive: bigint,
+    unpackEnvelopeOpts: UnpackEnvelopeOpts | undefined,
+): Promise<{ miniblocks: ParsedMiniblock[]; terminus: boolean }> {
+    const allMiniblocks: ParsedMiniblock[] = []
+    let currentFromInclusive = fromInclusive
+    let reachedTerminus = false
+
+    while (currentFromInclusive < toExclusive && !reachedTerminus) {
```

Comment:
1. shouldn't this be `currentFromInclusive <= toExclusive`? I would expect the next
2. you can't check both the condition and `reactedTerminus` if you're returning left to right

stream = [0,1,2,3,4,5]
maxMBs = 3
fetchMiniblocksFromRpc(0,6) will return { blocks: [0,1,2], reachedTerminus: true, fromInclusive: 0 }
// and the loop will exit

// for a trimmed stream
trimmedStream = [3,4,5,6,7,8]
maxMBs = 3
fetchMiniblocksFromRpc(0,6) will return { blocks: [3,4,5], reachedTerminus: true, fromInclusive: 3 }
// and the loop will exit

Reply:
🤦 yeah #1 is right as is, thanks for fixing #2